### PR TITLE
grouper/enricher produces/consumes keyed input

### PR DESCRIPTION
**Pull request by laxmanchekka**
## Description
grouper produces keyed output
enricher produces keyed input

### Testing
Unit tests

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
NA

### CODE REVIEW

**Review by kotharironak — APPROVED** [review]
lgtm

Can you check snyk issues?

File: hypertrace-trace-enricher/hypertrace-trace-enricher/src/test/java/HypertraceTraceEnricherTest.java [hunk 1]
```diff
@@ -7,18 +9,15 @@
 import java.util.Properties;
 import org.apache.kafka.common.serialization.Serde;
 import org.apache.kafka.common.serialization.Serdes;
-import org.apache.kafka.streams.StreamsBuilder;
-import org.apache.kafka.streams.TestInputTopic;
-import org.apache.kafka.streams.TestOutputTopic;
-import org.apache.kafka.streams.TopologyTestDriver;
+import org.apache.kafka.streams.*;
```
[the remark this answers is not in the document]
Comment (laxmanchekka) [reply hypertrace-trace-enricher/hypertrace-trace-enricher/src/test/java/HypertraceTraceEnricherTest.java:12]: done by spotless I guess. it prefers wildcard when number of classes used are more than a threshold

**Comment by laxmanchekka** [comment]
> lgtm
> 
> Can you check snyk issues?
@kotharironak : vulnerability is from data-model. we need to fix it separately and upgrade the lib here.

File: hypertrace-trace-enricher/hypertrace-trace-enricher/src/test/java/HypertraceTraceEnricherTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (saxenakshitiz) [reply hypertrace-trace-enricher/hypertrace-trace-enricher/src/test/java/HypertraceTraceEnricherTest.java:12]: Must be done by IDE. AFAIK spotless doesn't make this change.

[the remark this answers is not in the document]
Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher/src/test/java/HypertraceTraceEnricherTest.java:12]: I guess we need to set a high number like 999 as import class count to be optimised with `*` in intellij.

**Pull request merged**